src/generated/Admin/AdminRequestBuilder.cs

```diff
@@ -37,20 +38,29 @@ public class AdminRequestBuilder {
             };
             expandOption.IsRequired = false;
             command.AddOption(expandOption);
-            command.SetHandler(async (string[] select, string[] expand) => {
+            var outputOption = new Option<FormatterType>("--output", () => FormatterType.JSON){
+                IsRequired = true
+            };
+            command.AddOption(outputOption);
+            command.SetHandler(async (string[] select, string[] expand, FormatterType output, IConsole console) => {
+                var responseHandler = new NativeResponseHandler();
                 var requestInfo = CreateGetRequestInformation(q => {
                     q.Select = select;
                     q.Expand = expand;
                 });
-                var result = await RequestAdapter.SendAsync<ApiSdk.Models.Microsoft.Graph.Admin>(requestInfo);
+                await RequestAdapter.SendNoContentAsync(requestInfo, responseHandler);
                 // Print request output. What if the request has no return?
-                using var serializer = RequestAdapter.SerializationWriterFactory.GetSerializationWriter("application/json");
-                serializer.WriteObjectValue(null, result);
-                using var content = serializer.GetSerializedContent();
-                using var reader = new StreamReader(content);
-                var strContent = await reader.ReadToEndAsync();
-                Console.Write(strContent + "\n");
-            }, selectOption, expandOption);
+                var response = responseHandler.Value as HttpResponseMessage;
```

Comment:
this ties the generation output to a specific http client implementation. Please revert to using the serialization infrastructure kiota offers.

Reply:
Hey, thanks for pointing this out. Reverting to the serialization infrastructure will mean that the serialization issue on error responses will not be fixed until the error handling is resolved. Every endpoint that returns an error response will throw a serialization exception. I can move this over to core so that the generated code doesn't reference any http client implementation in the meantime.

Reply:
assume errors are working properly, once https://github.com/microsoft/kiota/issues/411 is implemented it won't be an issue anymore.